src/resources/prefabs.rs

```diff
@@ -1,48 +1,96 @@
 use std::collections::HashMap;
+use std::fs::read_dir;
 
 use amethyst::{
-    assets::{Handle, Prefab, PrefabLoader, RonFormat},
+    assets::{AssetStorage, Handle, Prefab, PrefabLoader, ProgressCounter, RonFormat},
     ecs::World,
+    utils::application_root_dir,
 };
 
-use crate::components::creatures::{CreaturePrefabData, CreatureType};
+use crate::components::creatures::CreaturePrefabData;
 
 #[derive(Default)]
 pub struct CreaturePrefabs {
-    prefabs: HashMap<CreatureType, Handle<Prefab<CreaturePrefabData>>>,
+    prefabs: HashMap<String, Handle<Prefab<CreaturePrefabData>>>,
 }
 
 impl CreaturePrefabs {
     pub fn insert(
         &mut self,
-        creature_type: CreatureType,
+        creature_type: String,
         prefab_handle: Handle<Prefab<CreaturePrefabData>>,
     ) {
         self.prefabs.insert(creature_type, prefab_handle);
     }
 
-    pub fn get_prefab(
-        &self,
-        creature_type: &CreatureType,
-    ) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
+    pub fn get_prefab(&self, creature_type: &str) -> Option<&Handle<Prefab<CreaturePrefabData>>> {
         self.prefabs.get(creature_type)
     }
+
+    pub fn get_prefabs(&self) -> &HashMap<String, Handle<Prefab<CreaturePrefabData>>> {
+        &self.prefabs
+    }
+
+    pub fn set_prefabs(&mut self, prefabs: HashMap<String, Handle<Prefab<CreaturePrefabData>>>) {
+        self.prefabs = prefabs;
+    }
 }
 
 // Here we load all prefabs for the different creatures in the game.
```

Comment:
Can you update the documentation for `initialize_prefabs` and `update_prefabs`, since it is using a workaround?

Reply:
Which documentation do you mean ? The dev log ? Or just comments in the code ?

Reply:
Just comments in the code, so that others browsing the code will know about the relationship between initialize and update